src/layout/GenericComponent.tsx

```diff
@@ -88,6 +89,27 @@ function ActualGenericComponent<Type extends CompTypes = CompTypes>({
   const validations = useUnifiedValidationsForNode(node);
   const isValid = !hasValidationErrors(validations);
   const isHidden = useIsHiddenComponent();
+  const { searchParams, clearSearchParam } = useNavigationParams();
+  const nodeId = searchParams.get(SearchParams.FocusNodeId);
+
+  const focusInput = React.useCallback(() => {
+    const maybeInput = containerDivRef.current?.querySelector('input,textarea,select,p') as
+      | HTMLSelectElement
+      | HTMLInputElement
+      | HTMLTextAreaElement;
+
+    if (maybeInput) {
+      maybeInput.focus();
+    }
+  }, []);
+
+  React.useEffect(() => {
+    if (nodeId === node.item.id) {
+      containerDivRef.current?.scrollIntoView();
+      focusInput();
+      clearSearchParam(SearchParams.FocusNodeId);
+    }
```

Comment:
This only works if the node is visible, so it wouldn't work if the node is in a repeating group. Look into `useNavigateToNode()` instead, which will open the group for editing, etc, before setting focus on it.

Also, with this method, the query parameter would linger in the URL until the node is suddenly visible, so you're not even guaranteed it will get focus when the page renders. Let's say you link to page `A` and nodeId `inPageB`. With this code the parameter would stay in the URL until you navigate to page `B` manually, at which point the `inPageB` node will suddenly (and unexpectedly) get focus.

Reply:
That's a very good point! I'll have to think about a better way to solve this :thinking:

Reply:
Yup, look into `useNavigateToNode()` and how to re-use that, which already solves the issue. Also, it should make sure to clear the param regardless of the focus request actually succeeding.

Reply:
I solved this by creating two different functions `linkToPage` and `linkToComponent`. Then, using `useNavigateToNode` solved the issues for navigating to a specific component.